File: sprint3/By languages/go/lesson7code1.go
package main

import "fmt"

func insertionSort(array []int) {
    for i := 1; i < len(array); i++ {
        itemToInsert := array[i]
        j := i
        for j > 0 && itemToInsert < array[j-1] {
            array[j] = array[j-1]
            j--
        }
        array[j] = itemToInsert
        fmt.Printf("step %d, sorted %d elements: %v\n", i, i+1, array)
    }
}

func main() {
    array := []int{4, 3, 2, 1}
    insertionSort(array)
}

File: sprint3/By languages/JS/lesson7code1.js
function insertionSort(array) {
  for (let i = 1; i < array.length; i++) {
    let item_to_insert = array[i];
    let j = i;
    while (j > 0 && item_to_insert < array[j - 1]) {
      array[j] = array[j - 1];
      j--;
    }
    array[j] = item_to_insert;
    console.log(`step ${i}, sorted ${i + 1} elements: ${array.slice(0, i + 1).join(" ")}`);
  }
}

let array = [5, 2, 4, 6, 1, 3];
insertionSort(array);
